networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/ActiveEth2Network.java

```diff
@@ -130,19 +148,53 @@ public ActiveEth2Network(
 
   @Override
   public SafeFuture<?> start() {
+    if (recentChainData.isPreGenesis() || recentChainData.isPreForkChoice()) {
+      throw new IllegalStateException(
+          getClass().getSimpleName()
+              + " should only be started after "
+              + recentChainData.getClass().getSimpleName()
+              + " is fully initialized.");
+    }
     // Set the current fork info prior to discovery starting up.
-    final ForkInfo currentForkInfo =
-        recentChainData
-            .getHeadForkInfo()
-            .orElseThrow(
-                () ->
-                    new IllegalStateException("Can not start Eth2Network before genesis is known"));
+    final ForkInfo currentForkInfo = recentChainData.getHeadForkInfo().orElseThrow();
     discoveryNetwork.setForkInfo(currentForkInfo, recentChainData.getNextFork());
     return super.start().thenAccept(r -> startup());
   }
 
   private synchronized void startup() {
     state.set(State.RUNNING);
+    queueGossipStart();
+  }
+
+  private void queueGossipStart() {
+    LOG.debug("Check if gossip should be started");
+    final UInt64 slotsBehind = recentChainData.getChainHeadSlotsBehind().orElseThrow();
+    if (slotsBehind.isLessThanOrEqualTo(500)) {
+      // Start gossip if we're "close enough" to the chain head
+      // Note: we don't want to be too strict here, otherwise we could end up with our sync logic
+      // inactive because our chain is almost caught up to the chainhead, but gossip inactive so
+      // that our node slowly falls behind because no gossip is propagating.  However, if we're too
+      // aggressive, our node could be down-scored for subscribing to topics that it can't yet
+      // validate causing our node to fail to propagate any gossip.
+      startGossip();
+    } else {
+      // Check again when we should be caught up assuming a speedy sync process
+      final int blocksPerSecond = 100;
+      final int delayInSeconds = slotsBehind.dividedBy(blocksPerSecond).min(600).max(10).intValue();
+      LOG.debug(
+          "Chain is not yet in sync, check if gossip should be started in {} seconds",
+          delayInSeconds);
```

Comment:
I wonder if we're trying to be too clever here and should just retry after a fixed amount of time. It's a pretty cheap check to perform.

Reply:
Maybe a bit much - although it's kind of fun, right? :D Just seemed excessive to check every 5 seconds or whatever in the case where we're going to be syncing for hours.  But I can simplify.